core/src/main/java/org/apache/spark/TaskContext.java

```diff
@@ -94,6 +94,14 @@ static void unset() {
   @Deprecated
   public abstract void addOnCompleteCallback(final Function0<Unit> f);
 
+  /**
+   * Add a callback function to be executed on task stop.
+   * This will be called when task is being killed.
+   *
+   * @param f Callback function.
+   */
+  public abstract void addOnStopCallback(final Function1<String, Unit> f);
```

Comment:
I think this name is potentially confusing, since users might get confused about the difference between "stop" and "completion".  I'd prefer something like "onKill" to make it more obvious that this is only called for killed tasks.

Also, what is the string argument that's passed in to the callback?

Also, this seems similar to the deprecated `addOnCompleteCallback`.  Maybe we should fold the new callback into `TaskCompletionListener` instead?